### DISCUSSION TITLE
Is there any plan for supporting `Vue` and `Solid`?

### DISCUSSION THREAD

**Discussion by xsjcTony** [xyflow/xyflow#3761] [New Features]
Now we already have an official `Svelte` version, which is ✨impressive.

Wondering whether there's any plan for supporting Vue and Solid?

- `Vue` is still relatively a larger (by usage amount and ecosystem) framework than `Svelte`, I was actually shocked to see that there's a Svelte port but no Vue's.
- `Solid` seems to be the best-performant framework at the moment and it may (I believe) become a giant in the future.

Thanks.

**Comment by bcakmakoglu** [comment]
You might wanna check out https://vueflow.dev if you're interested in a Vue version of React Flow ^^

It's not as neatly documented as React Flow, though basically anything that would work with v11 of React Flow works with Vue Flow as well.

**Reply by xsjcTony** [reply]
Just double-checking it's non-official right?

**Reply by xsjcTony** [reply]
I don't mean anything bad, but just need to take that into consideration when picking libraries 👍

**Reply by bcakmakoglu** [reply]
It's not officially maintained by the React Flow team, that's right.

Though it's used by quite a few projects and not just 0 star ones :)
I understand it'd be cool if there was `@xyflow/vue` (which maybe will happen? Who knows 😉) though.

**Reply by xsjcTony** [reply]
Cheers mate

**Reply by hayleigh-dot-dev** [reply]
We can definitely recommend Vue Flow if you're looking for a Vue alternative :) @bcakmakoglu is great!

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by JoelCodes** [comment]
Yeah, I'd definitely be here for a Solid version.  Is there a process for adding versions?  Like, we vote on a version, a branch or set of branches is created, PR's are submitted, etc.?

Maybe a proposal?

**Comment by asterikx** [comment]
Also very interested in a Solidjs version!

**Comment by yoavbls** [comment]
I'm here too for the solid version ✌🏽

**Comment by warrenbhw** [comment]
solidJS +1

**Comment by CSworenI** [comment]
@xsjcTony @JoelCodes @yoavbls @asterikx @warrenbhw 
Perhaps you guys haven't heard of [solid-flow](https://github.com/miguelsalesvieira/solid-flow). As with the mentioned vueflow, it's not maintained by the xyflow/reactflow team, but it is out there and open source.

I would also be in favor of the xyflow team creating a professional package, though...

**Comment by yoavbls** [comment]
@CSworenI I see the last commit in solid-flow was 2 years ago. It'll be so nice to have something official

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by Isaak-Marcel** [comment]
I would also love and need a solidJS official version of this.

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by warrenbhw** [comment]
A friend of mine has already written the solid port of xyflow:
https://www.npmjs.com/package/@matthewgapp/solidjs-flow

https://github.com/nobie-org/xyflow/pull/1

He put up a PR at one point to merge it into the main Xyflow codebase, but I don't think he got any response from code reviewers so he ended up closing it.

But that npm package should work

**Reply by warrenbhw** [reply]
cc @matthewgapp

**Reply by matthewgapp** [reply]
@xsjcTony if someone wants to pick up my port or use it as the basis for something that will get merged here, that would be great. We're using it within our application and seems to be working fine :) 

Just haven't had time to upstream it.

**Reply by agoldstein03** [reply]
The port looks awesome but also (of course understandably) a bit out of date. +1 for official upstream SolidJS support, if the financial incentives would help progression definitely let us know if there is a spot to indicate "I would subscribe if this feature were implemented", or even a bounty or something.

Also, with official Svelte 5 support released a few weeks ago, chances are creating some shared vanilla JS minimal shared libs could probably help with SolidJS support and Svelte 5 runes at the same time, since they are both essentially signal-based 🙂

**Reply by xsjcTony** [reply]
That is a very valid point. A shared vanilla JS core can help a lot. Basically, it's the React vs whole signal world right now (even Angular is using signal).

With the official Svelte 5 support, there should be some parts that can be extracted and shared across frameworks which can help other signal-based frameworks I believe.

(Honestly Vue is the one starting the Signal architecture first but everyone talks about Solid nowadays for inventing signal patterns😂)

**Reply by matthewgapp** [reply]
A shared core would help. But I don't think that project should block a solid port. 

If someone has a day to throw at porting it over, it's not hard - I did most of above port in an afternoon. Just swap out all react state for signals. I wrapped solid signals in .get(), .set() interface to make it more similar.

**Reply by bcakmakoglu** [reply]
There *is* a shared core called `@xyflow/system`

**Reply by matthewgapp** [reply]
Right! That's what I used when I ported, which made it easier. Even more shared code would be helpful as there was still a lot of state code to change but, like I said, I don't think that should block a port.

**Reply by matthewgapp** [reply]
I'm happy to refresh my port one afternoon if the core team will commit to merging it. Otherwise, I'll fork this library and maintain a standalone Solid JS variant similar to https://vueflow.dev/

**Reply by JasonWeise** [reply]
@matthewgapp , did you have a repo for solid-flow, all the links I find seem to link back to xyflow/xyflow and not an actual solidjs version (or am I missing something)
I'd love a SolidJS version as it is something definitely missing for Sold.

**Reply by brainless** [reply]
This port looks nice :star_struck:, thanks @matthewgapp.

I am curious how much of an ongoing effort will it be to switch between React + zustand and Solid way of doing things? Would be better to maintain a spiritual clone/sibling? Yes, then the features may diverge at some point but I think the visible features can stay the same while still writing directly for Solid and not have to translate state/rendering logic.

@JasonWeise here is the repo: https://github.com/nobie-org/xyflow

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by oddpxl** [comment]
SolidJS +1 

Please :)

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by yefu24324** [comment]
SolidJS +1

**Reply by EzzatOmar** [reply]
SolidJS +1

**Reply by chalapathi-ct** [reply]
SolidJS +1

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by paulm17** [comment]
@matthewgapp A bit OT.  But could you speak about the challenges from porting a library from react?  I'm considering moving from one to the other but finding that the solidjs ecosystem just isn't where it should be.  I don't mind getting my hands dirty (so to speak).  Are there some things that I should be aware of?  Or perhaps some techniques or things I should know before hand?  Thanks

**Reply by matthewgapp** [reply]
No challenges really, @paulm17. Didn't take longer than a day. Mostly rote work that could be done by Cursor now. Porting is easy. Getting it upstreamed and maintained is the important bit.

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by matthewgapp** [comment]
@EzzatOmar, @paulm17 @chalapathi-ct @oddpxl @JasonWeise @brainless @bcakmakoglu @yoavbls @CSworenI 

Perhaps one of you could bring my port up to date and then we can get it upstreamed into xyflow :). There doesn't seem to be that many changes between the main of my port and the main of this repo. 

Changes here: https://github.com/nobie-org/xyflow/compare/main...xyflow%3Axyflow%3Amain

90% of the changes are doc-comments. 

Does anyone want to raise their hand and be a hero? 🤚

**Reply by paulm17** [reply]
I have a definite need for xyflow in an extensive react application that I am building.  Unfortunately I'm tied up in rust land at the moment.  Additionally, I have decided to delay my foray into solidjs until 2.0 and HMR gets fixed.  When that happens, if no one has upstreamed xyflow to solidjs then I'll definitely take a stab at it.

**Reply by matthewgapp** [reply]
hey @EzzatOmar @paulm17 @chalapathi-ct @oddpxl @JasonWeise @brainless @bcakmakoglu @yoavbls @CSworenI @JoelCodes @hayleigh-dot-dev @asterikx @warrenbhw @yefu24324 @Isaak-Marcel @agoldstein03 @xsjcTony

Any takers? You'd be a hero. 

If we can get this in shape (which shouldn't take much work since most of the changes required are doc comments), I’m sure we can make this official.

Repo: https://github.com/nobie-org/xyflow
NPM: https://www.npmjs.com/package/@matthewgapp/solidjs-flow
Diff: nobie-org/xyflow@main...xyflow:xyflow:main

**Reply by paulm17** [reply]
@matthewgapp 
I might be able to.  I'm porting over mantine from react to solid.  So I've already started the learning process.  https://github.com/paulm17/mantinesolid

Let's see how confident I feel when I have finished. 😓

**Reply by matthewgapp** [reply]
Good stuff. fwiw, updating my port will be a lot easier than porting over mantine.

**Reply by matthewgapp** [reply]
FYI https://github.com/xyflow/xyflow/discussions/3761#discussioncomment-13267913

**Comment by matthewgapp** [comment]
hey @EzzatOmar @paulm17 @chalapathi-ct @oddpxl @JasonWeise @brainless @bcakmakoglu @yoavbls @CSworenI @JoelCodes @hayleigh-dot-dev @asterikx @warrenbhw @yefu24324 @Isaak-Marcel @agoldstein03 @xsjcTony - 

I updated my port and created a [PR](https://github.com/xyflow/xyflow/pull/5293) should be good to get merged in once I test a bit more. 

🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀

**Reply by matthewgapp** [reply]
Looks like there's no immediate future for SolidJS here, so I'll be publishing a fork shortly https://github.com/xyflow/xyflow/pull/5293#issuecomment-2911454399

**Reply by thefrana** [reply]
@matthewgapp thank you for work and your willingness to push it forward!

**Reply by paulm17** [reply]
@matthewgapp same, thank you for your work!  This is really appreciated!

**Reply by QWu4xYV** [reply]
@matthewgapp Thanks so much for all your work on this! Do you know when the fork will be published? Excited to try it :)

**Reply by asterikx** [reply]
@matthewgapp Thanks so much! Is there a possibility to release it as a package?

**Comment by oddpxl** [comment]
@matthewgapp Hey - sorry to chase - just curious if you still plan to fork ? 👍 👍  ?

**Comment by paulm17** [comment]
@matthewgapp I actually completed the mantine fork.  Although I don't need this right now.  Do you need any help with anything or any updates to come?

**Comment by thedanchez** [comment]
Hey all, if any are still interested in Solid Flow, I just recently published this that is now available for use: https://www.npmjs.com/package/@dschz/solid-flow

I consider this an alpha release with a couple of things missing that are outlined in the README, but a lot should be there available for use. Hoping it will address the majority of use cases. I hope it helps!

**Reply by yefu24324** [reply]
Support you, I'll try to create a new branch with this

**Reply by Borber** [reply]
Thank you for your work, I will try to use it 💖
